lib/src/main/kotlin/ca/aqtech/mctreesearch4j/Node.kt

```diff
@@ -1,19 +1,14 @@
 package ca.aqtech.mctreesearch4j
 
-abstract class Node<TAction, TSelf>(
-    val parent: TSelf?,
-    val inducingAction: TAction?
-) where TSelf : Node<TAction, TSelf> {
-    val depth: Int = if (parent == null) {
-        0
-    }
-    else {
-        parent.depth + 1
-    }
+abstract class Node<ActionType, SelfType: Node<ActionType, SelfType>> (
```

Comment:
Turns out `TType` is not the standard convention for naming type parameters. Although the Java convention is singular capital letters, I find that terribly confusing so I've opted for the alternative which is the *Type suffix.

For reference: https://docs.oracle.com/javase/tutorial/java/generics/types.html